misk-api/src/main/kotlin/misk/scope/Scope.kt

```diff
@@ -1,9 +1,11 @@
 package misk.scope
 
+import kotlin.reflect.KType
+
 interface Scope : AutoCloseable {
-  fun <T> get(type: Class<T>): T?
+  fun <T> get(key: KType): T
```

Comment:
Have tests been done that this will support annotations or other Guice like qualifiers on types?

Reply:
Yes, I tested with different Kotlin types to see if we can replace Guice but none of them actually worked as expected. For now, reverted to using Guice in misk-api.